Allow usage of ESLint v7

Current version of `eslint-plugin-aura` has an outdated dependency on `eslint-plugin-compat@3.3.0` which doesn't support the latest major version of ESLint (v7 which is close to a year old).

This results in broken builds:
```
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! Found: eslint@7.20.0
npm ERR! node_modules/eslint
npm ERR!   dev eslint@"^7.20.0" from the root project
npm ERR!   peer eslint@"^6 || ^7" from @salesforce/eslint-config-lwc@0.9.0
npm ERR!   node_modules/@salesforce/eslint-config-lwc
npm ERR!     dev @salesforce/eslint-config-lwc@"^0.9.0" from the root project
npm ERR!   6 more (@salesforce/eslint-plugin-aura, ...)
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer eslint@"^3.0.0 || ^4.0.0 || ^5.0.0 || ^6.0.0" from eslint-plugin-compat@3.3.0
npm ERR! node_modules/@salesforce/eslint-plugin-aura/node_modules/eslint-plugin-compat
npm ERR!   eslint-plugin-compat@"3.3.0" from @salesforce/eslint-plugin-aura@1.4.0
npm ERR!   node_modules/@salesforce/eslint-plugin-aura
npm ERR!     dev @salesforce/eslint-plugin-aura@"^1.4.0" from the root project
```

--- a/package.json
+++ b/package.json
@@ -37,7 +37,7 @@
     "prettier": "^2.2.1"
   },
   "peerDependencies": {
-    "eslint": ">=5.0.0"
+    "eslint": "^7"
   },
   "dependencies": {
     "eslint-plugin-compat": "^3.9.0"

